main.go

```diff
@@ -109,54 +117,137 @@ func runCmdFunc(cmd *cobra.Command, args []string) error {
 			return fmt.Errorf("not enough arguments provided")
 		}
 		program = args[0]
-		pathArgs = args[1:]
+		paths = args[1:]
 	}
 
 	// Handle each file path provided.
-	for _, pathArg := range pathArgs {
-		err := processPathArg(pathArg, program, flags)
+	if flags.slurp {
+		if flags.outputFormat == "" {
+			return fmt.Errorf("must specify --output-format when using --slurp")
+		}
+		encoder, ok := formatByName(flags.outputFormat)
+		if !ok {
+			return fmt.Errorf("invalid --output-format %s", flags.outputFormat)
+		}
+		err := slurpFiles(paths, program, encoder, flags)
+		if err != nil {
+			return err
+		}
+	} else {
+		err := processFiles(paths, program, flags)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// processFiles takes a list of files, and for each, attempts to convert it
+// to a JSON value and runs the jq program it
+func processFiles(paths []string, program string, flags flags) error {
+	for _, path := range paths {
+		fileInfo, err := readFile(path, flags)
+		if err != nil {
+			return err
+		}
+
+		err = runJQ(program, fileInfo.data, fileInfo.encoder, flags)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// slurpFiles takes a list of files, and for each, attempts to convert it to
+// a JSON value and appends each JSON value to an array, and passes that array
+// as the input to the jq program.
+func slurpFiles(paths []string, program string, encoder formats.Encoding, flags flags) error {
+	// we ignore the errors because byte.Buffers generally do not return
+	// error on write, and instead only panic when they cannot grow the
+	// underlying slice.
+	var buf bytes.Buffer
+
+	// append the first array bracket
+	buf.WriteRune('[')
+
+	// iterate over each file, appending it's contents to an array
+	for i, path := range paths {
+		fileInfo, err := readFile(path, flags)
 		if err != nil {
 			return err
 		}
+		// only handle files with content
+		if len(bytes.TrimSpace(fileInfo.data)) != 0 {
+			buf.Write(fileInfo.data)
+			// append the comma if it isn't the last item in the array
+			if i != len(paths)-1 {
+				buf.WriteRune(',')
+			}
+		}
+	}
+	// apend the last array bracket
+	buf.WriteRune(']')
+
+	data := buf.Bytes()
+	if len(data) == 0 {
```

Comment:
I don't think this is ever possible because we're writing `[` and `]` regardless.

Reply:
Good point. And it seems the jq behavior is that too:

```
echo -n '' | jq -s '.'
[]
```